File: 01 introduction/pesquisa_binaria.js
// Method development test binary search
const pesquisaBinaria = (lista, item) => {
  let baixo = 0;
  let alto = lista.length - 1;

  while (baixo <= alto) {
    const meio = Math.floor((baixo + alto) / 2);
    const chute = lista[meio];

    if (chute == item) {
      return meio
    }
    else if (chute > item) {
      alto = meio - 1
    }
    else {
      baixo = meio + 1
    }
  }
  return null
};

const minha_lista = [1, 3, 5, 7, 9];

console.log(pesquisaBinaria(minha_lista, 3)); // 1
console.log(pesquisaBinaria(minha_lista, -1)); // nil

File: 01 introduction/pesquisa_binaria.go
// Method development test binary search
package main

import "fmt"

func pesquisaBinaria(lista []int, item int) bool {
	baixo := 0
	alto := len(lista) - 1

	for baixo <= alto {
		meio := (baixo + alto) / 2

		if lista[meio] == item {
			return true
		}
		if lista[meio] > item {
			alto = meio - 1
		} else {
			baixo = meio + 1
		}
	}
	return false
}

func main() {
	fmt.Println(pesquisaBinaria([]int{1, 2, 3, 4, 5}, 1))  // true
	fmt.Println(pesquisaBinaria([]int{1, 2, 3, 4, 5}, -1)) //false
}
